### PR TITLE
Next/previous

### PR DESCRIPTION
**Pull request by Xon**
Next/Previous buttons.

Includes:
- piles of mobile tweaks to smooth out the next/previous buttons.
- news-feed support
- Search support (still missing the threadmark specific tab)
- edit history

### CODE REVIEW

**Comment by Xon** [comment]
latest css apparently causes crashes in some version of opera mobile. figuring out why is going to be challenging as I don't have access to that browser.

**Comment by Xon** [comment]
This branch is being used on SpaceBattles, and has ~20k threadmarks in use. While SufficientVelocity has 26k threadmarks in use on a slightly different version.

**Comment by Xon** [comment]
@sidane over a year later, are you still interested in merging in any of my changes or should I fork this completely in XF Resource Manager?

**Comment by sidane** [comment]
Apologies @Xon, yes I do want to get these changes added. I have free time on Friday so I'll review this and #44 then.

**Comment by Xon** [comment]
@sidane there is about ~39 commits difference between the head of my fork and yours.

The next major feature (besides a ton of bugfixes and cleanup) is "Sortable Threadmarks".

File: upload/library/Sidane/Threadmarks/Model/Post.php [hunk 1]
```diff
@@ -14,12 +15,28 @@ public function preparePostJoinOptions(array $fetchOptions)
...
-          threadmarks.threadmark_id, threadmarks.label as threadmark_label
+          threadmarks.threadmark_id, threadmarks.label as threadmark_label, threadmarks.edit_count as threadmark_edit_count,
+          threadmarks.user_id as threadmark_user_id, threadmarks.position as threadmark_position
         ';
         $joinOptions['joinTables'] .= '
           LEFT JOIN threadmarks  ON
             threadmarks.post_id = post.post_id
         ';
+
+        //if (($fetchOptions['join'] & Sidane_Threadmarks_Model_Post::FETCH_THREADMARKS_FULL) == Sidane_Threadmarks_Model_Post::FETCH_THREADMARKS_FULL)
```
Comment (sidane) [inline upload/library/Sidane/Threadmarks/Model/Post.php]: Can this commented code be deleted?

File: upload/library/Sidane/Threadmarks/Search/DataHandler/Threadmark.php [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+
+    if ($parentData)
+    {
+      $thread = $parentData;
+      if ($thread['discussion_state'] != 'visible')
+      {
+        return;
+      }
+      // title only makes sense if a collection of threadmarks have their own title independent of threads
+      //$title = $thread['title'];
```
Comment (sidane) [inline upload/library/Sidane/Threadmarks/Search/DataHandler/Threadmark.php]: Can these two lines be removed?

**Comment by sidane** [comment]
Tested and verified locally. Just a few minor comments regarding commented out code.

More nice additions, I particularly like the search integration.

One request: Can the next/previous links be optional via a configuration in the Admin CP? It's not a feature I will use on my forum so would like to be able to turn it off.

**Comment by Xon** [comment]
Implemented the feedback.

@sidane you happy with it just being a tiny template edit? If so the most recent commit should be good :)

**Comment by sidane** [comment]
Thanks for that, happy with the template change. Apologies though, I also meant the 'Index' link should be optional as well. I wasn't clear on that at all 😞 

Just use the same Admin CP feature flag I think.

Can you also please rebase this branch on master to remove the commits from #44?

**Comment by Xon** [comment]
Rebased, and made the entire Threadmark nav bar to be optional.

**Comment by sidane** [comment]
Thanks @Xon but I think the 'Index' link is still always visible?

I'm travelling the next few days so will get this merged and a new version of the add-on released this weekend.

**Comment by Xon** [comment]
Cheers.

@sidane I'm not sure how it can be visible, as I wrapped the entire contents of threadmarks_nav in an xen:if statement. I didn't change the verison info, so sometimes XF doesn't recompile the changes properly.

There is actually another batch of commits after this which implements sortable threadmarks for the index list.

**Comment by sidane** [comment]
You're right about the index, my bad.

Ok I'll merge this now so you can open the next pull request.

**Comment by Xon** [comment]
Should have the next PR done quickly

**Pull request merged**